src/Lachain.Core/RPC/HTTP/HttpService.cs

```diff
@@ -179,43 +194,70 @@ private bool _Handle(HttpListenerContext context)
-                        {
-                            serializedParams += param.Key + param.Value?.ToString();
-                        }
-                    }
-                }
+                    if(string.IsNullOrEmpty(signature)) return false;
+                    if(string.IsNullOrEmpty(timestamp)) return false;
+                    // If unix timestamp diff is longer than 30 minutes, we dont handle it
+                    if(!long.TryParse(timestamp.Trim(), out long unixTimestamp)) return false;
+                    TimeSpan timeSpan = DateTimeOffset.Now.Subtract(DateTimeOffset.FromUnixTimeSeconds(unixTimestamp));
+                    if(timeSpan.TotalMinutes >= 30)
+                        return false;
 
-                var messageToSign = body["method"]!.ToString() + serializedParams + timestamp;
-                var messageBytes = Encoding.UTF8.GetBytes(messageToSign);
-                var messageHash = System.Security.Cryptography.SHA256.Create().ComputeHash(messageBytes);
+                    // Serialize params (format: key1value1key2value2...)
+                    string serializedParams = SerializeParams(body["params"] as JObject);
 
-                var signatureBytes = signature.HexToBytes();
-                var publicKey = _apiKey!.HexToBytes();
+                    var messageToSign = body["method"]!.ToString() + serializedParams + timestamp;
+                    var messageBytes = Encoding.UTF8.GetBytes(messageToSign);
+                    Logger.LogTrace($"Meesage to sign: {messageBytes.ToHex()}");
+                    var messageHash = Crypto.HashUtils.KeccakBytes(messageBytes);
+                    Logger.LogTrace($"Meesage hash: {messageHash.ToHex()}");
+                    
+                    Logger.LogTrace($"API public key: {_apiKey}");
 
-                var secp256K1 = new Secp256k1();
-                return secp256K1.Verify(signatureBytes, messageHash, publicKey);
+                    var secp256K1 = new Secp256k1();
+                    var sigBytes = signature.HexToBytes();
+                    if (sigBytes.Length != 65)
+                        throw new Exception("Invalid signature length");
+                    var parsedSig = new byte[65];
+                    var pk = new byte[64];
+                    var recId = sigBytes[64];
+                    if (recId < 0 || recId > 3)
+                        throw new Exception($"Bad signature,  invalid recId={recId}: : recId >= 0 && recId <= 3 ");
+                    if (!secp256K1.RecoverableSignatureParseCompact(parsedSig, sigBytes.Take(64).ToArray(), recId))
+                        throw new ArgumentException(nameof(signature));
+                    if (!secp256K1.Recover(pk, parsedSig, messageHash))
+                        throw new ArgumentException("Bad signature");
+                    Logger.LogTrace($"Recovered public key: {pk.ToHex()}");
+                    return pk.ToHex() == _apiKey!;
```

Comment:
This match could be wrong in case of CloudKeygen  because the public key will be 33 bytes but pk is 64 bytes. We should generate the compressed one from pk